official/resnet/ctl/ctl_imagenet_main.py

```diff
@@ -0,0 +1,297 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Runs a ResNet model on the ImageNet dataset using custom training loops."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import tempfile
+import datetime
+import time
+
+from absl import app as absl_app
+from absl import flags
+# from absl import logging
+
+import tensorflow.compat.v1.logging as logging
+import tensorflow as tf
+import numpy as np
+
+from official.resnet import imagenet_main
+from official.resnet.keras import keras_common
+from official.resnet.keras import resnet_model
+from official.utils.flags import core as flags_core
+from official.utils.logs import logger
+from official.utils.misc import distribution_utils
+from official.utils.misc import model_helpers
+from official.resnet.ctl import ctl_common
+
+
+# Imagenet training and test data sets.
+APPROX_IMAGENET_TRAINING_IMAGES = 1280000  # Approximate number of images.
```

Comment:
Can we move this into keras_common or some common resnet module, even the other main.py?  I do not want to put code clean up on you but I suspect we have this value and the other values below in many locations.

Reply:
I think Keras uses imagenet_main so I've done the same for these benchmarks as well. We should move this into a common repo. Right now we need a resnet_common.py which can be shared by keras, ctl and estimator.